server.py

```diff
@@ -5,79 +5,158 @@
     except Exception as e:
-        return jsonify({"status": "error", "message": str(e)}), 500
+        return ErpResponse(False, str(e), status_code=500).to_response()
+
 
 @app.route("/login", methods=["POST"])
-def login_and_download_ics():
+def login():
     try:
-        token = request.headers.get("Authorization")
-        email_otp = request.form.get("email_otp")
-        
-        if not all([token, email_otp]):
-            return jsonify({"status": "error", "message": "Missing token or email OTP"}), 400
-        
-        session_manager.establish_erp_session(token, email_otp)
-        return jsonify({"status": "success"}), 200
+        jwt = None
+        auth_resp, status_code = handle_auth()
+        if status_code != 200:
+            return auth_resp, status_code
+        else:
+            jwt = auth_resp.get_json().get("jwt")
+
+        password = request.form.get("password")
+        secret_answer = request.form.get("secret_answer")
+        otp = request.form.get("otp")
+        if not all([secret_answer, password, otp]):
+            return ErpResponse(False, "Missing password, secret answer or otp", status_code=400).to_response()
+
+        session_manager.login(jwt, password, secret_answer, otp)
+        return ErpResponse(True, message="Logged in to ERP").to_response()
     except Exception as e:
-        return jsonify({"status": "error", "message": str(e)}), 500
+        return ErpResponse(False, str(e), status_code=500).to_response()
+
+
+@app.route("/logout", methods=["GET"])
+def logout():
+    try:
+        jwt = None
+        auth_resp, status_code = handle_auth()
+        if status_code != 200:
+            return auth_resp, status_code
+        else:
+            jwt = auth_resp.get_json().get("jwt")
 
+        session_manager.end_session(jwt=jwt)
+
+        return ErpResponse(True, message="Logged out of ERP").to_response()
+    except Exception as e:
+        return ErpResponse(False, str(e), status_code=500).to_response()
+
+
+    
 @app.route("/download-ics", methods=["POST"])
 def download_ics():
     try:
-        token = request.headers.get("Authorization")
+        jwt = None
+        auth_resp, status_code = handle_auth()
+        if status_code != 200:
+            return auth_resp, status_code
+        else:
+            jwt = auth_resp.get_json().get("jwt")
         
-        if not token:
-            return jsonify({"status": "error", "message": "Missing token"}), 400
+        _, ssoToken = session_manager.get_erp_session(jwt=jwt)
         
-        session = session_manager.get_erp_session(token)
-        erp_session = ERPSession.create_erp_session(session.requests_session, session.sso_token, session.login_details.get("user_id"))
+        data = request.form
+        roll_number = data.get("roll_number")
+         
+        session = requests.Session()
+        erp_session = ERPSession.create_erp_session(session, ssoToken,roll_number)
```

Comment:
No.
All this will change.
We already have a method in our session manager to make GET/POST/PUT/DELETE requests.

You may also have to change the internal architecture of GYFT a bit